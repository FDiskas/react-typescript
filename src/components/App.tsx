import * as React from "react";

import { Demo } from "./Demo";

export interface IAppProps {
    title: string;
}

export class App extends React.Component<IAppProps, {}> {
    public render() {
        return <h1>Hello... <Demo title={this.props.title}/>from {this.props.title}!</h1>;
    }
}
