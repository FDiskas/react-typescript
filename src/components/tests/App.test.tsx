import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';
import * as React from 'react';

import { App } from '../App';

describe('App.js', () => {
    it('renders without crashing', () => {
        shallow(<App title="Test" />);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<App title="Test" />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('Title should be visible', () => {
        const wrapper = shallow(<App title="Test" />);
        expect(wrapper.find('h1').text()).toContain('Test');
    });

    it('Title should not be visible', () => {
        const wrapper = shallow(<App />);
        expect(wrapper.find('h1').text().length).toBe(0);
    });

    it('should handle click', () => {
        const wrapper = shallow(<App />);
        const params = { id: 1, label: 'Demo' };
        wrapper.instance().handleClick(params, true);

        expect(wrapper.instance().selectedItems.has(1)).toBeTruthy();

        wrapper.instance().handleClick(params, false);

        expect(wrapper.instance().selectedItems.has(1)).toBeFalsy();
    });
});
