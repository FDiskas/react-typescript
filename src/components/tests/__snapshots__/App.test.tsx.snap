// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js matches snapshot 1`] = `
<div>
  <h1>
    Test
  </h1>
  <CheckBox
    checked={true}
    handleClick={[Function]}
    item={
      Object {
        "id": 1,
        "label": "Foo",
      }
    }
  />
  <CheckBox
    checked={true}
    handleClick={[Function]}
    item={
      Object {
        "id": 2,
        "label": "Buzz",
      }
    }
  />
  <CheckBox
    checked={false}
    handleClick={[Function]}
    item={
      Object {
        "id": 3,
        "label": "Fuzz",
      }
    }
  />
  <CheckBox
    checked={false}
    handleClick={[Function]}
    item={
      Object {
        "id": 4,
        "label": "Fizz",
      }
    }
  />
  <CheckBox
    checked={false}
    handleClick={[Function]}
    item={
      Object {
        "id": 5,
        "label": "FizzBuzz",
      }
    }
  />
  <CheckBox
    checked={true}
    handleClick={[Function]}
    item={
      Object {
        "id": 6,
        "label": "Boo",
      }
    }
  />
</div>
`;
